daft/expressions/visitor.py

```diff
@@ -151,3 +153,171 @@ def visit_is_null(self, expr: Expression) -> R:
     def visit_not_null(self, expr: Expression) -> R:
         """Visit an not_null predicate."""
         ...
+
+
+class _PyArrowExpressionVisitor(PredicateVisitor[pc.Expression]):
+    """This visitor does a tree fold into the pyarrow.compute.Expression domain."""
+
+    def visit_col(self, name: str) -> pc.Expression:
+        """Convert the daft column to pc field reference by name."""
+        return pc.field(name)
+
+    def visit_lit(self, value: Any) -> pc.Expression:
```

Comment:
Just curious, do we support more complex scalars? Like datetime / date / timestamp, lists, structs, etc? Are they reduced in earlier steps?

Reply:
Yes, these are converted to the appropriate python types in the rust to_pyobject implementation.